Test failures on Jenkins Windows build

```

======================================================================
ERROR: Failure: ImportError (No module named 'terminado')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\ADMINI~1\AppData\Local\Temp\2\28634\lib\site-packages\nose\failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "C:\Users\ADMINI~1\AppData\Local\Temp\2\28634\lib\site-packages\nose\loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "C:\Users\ADMINI~1\AppData\Local\Temp\2\28634\lib\site-packages\nose\importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "C:\Users\ADMINI~1\AppData\Local\Temp\2\28634\lib\site-packages\nose\importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "C:\Users\ADMINI~1\AppData\Local\Temp\2\28634\lib\imp.py", line 244, in load_module
    return load_package(name, filename)
  File "C:\Users\ADMINI~1\AppData\Local\Temp\2\28634\lib\imp.py", line 216, in load_package
    return _load(spec)
  File "<frozen importlib._bootstrap>", line 693, in _load
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 665, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "C:\jenkins\workspace\notebook-windows\_pyversion\3.5\label\windows\notebook\terminal\__init__.py", line 3, in <module>
    import terminado
ImportError: No module named 'terminado'
-------------------- >> begin captured logging << --------------------
NotebookApp: INFO: Shutting down kernels
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: Should copy select files from source to destination
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\jenkins\workspace\notebook-windows\_pyversion\3.5\label\windows\notebook\bundler\tests\test_bundler_tools.py", line 122, in test_copy_filelist
    self.assertTrue(os.path.isfile(os.path.join(self.tmp, 'resources/subdir/test_file.txt')))
AssertionError: False is not true
-------------------- >> begin captured logging << --------------------
NotebookApp: INFO: Shutting down kernels
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: Should expand to single file in the resources/ subfolder.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\jenkins\workspace\notebook-windows\_pyversion\3.5\label\windows\notebook\bundler\tests\test_bundler_tools.py", line 88, in test_glob_dir
    tools.expand_references(HERE, ['resources/empty.ipynb']))
AssertionError: 'resources/empty.ipynb' not found in {'resources\\empty.ipynb'}

======================================================================
FAIL: Should expand to test_file.txt deep under this test/ directory.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\jenkins\workspace\notebook-windows\_pyversion\3.5\label\windows\notebook\bundler\tests\test_bundler_tools.py", line 104, in test_glob_splatsplat_in_middle
    self.assertIn('resources/subdir/test_file.txt', globs, globs)
AssertionError: 'resources/subdir/test_file.txt' not found in set() : set()

======================================================================
FAIL: Should expand to test_file.txt under any path.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\jenkins\workspace\notebook-windows\_pyversion\3.5\label\windows\notebook\bundler\tests\test_bundler_tools.py", line 115, in test_glob_splatsplat_leading
    self.assertIn('resources/subdir/test_file.txt', globs, globs)
AssertionError: 'resources/subdir/test_file.txt' not found in set() : set()

======================================================================
FAIL: Should expand to all descendants of this test/ directory.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\jenkins\workspace\notebook-windows\_pyversion\3.5\label\windows\notebook\bundler\tests\test_bundler_tools.py", line 109, in test_glob_splatsplat_trailing
    self.assertIn('resources/empty.ipynb', globs, globs)
AssertionError: 'resources/empty.ipynb' not found in set() : set()

======================================================================
FAIL: Should expand to all files in the resources/ subfolder.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\jenkins\workspace\notebook-windows\_pyversion\3.5\label\windows\notebook\bundler\tests\test_bundler_tools.py", line 93, in test_glob_subdir
    tools.expand_references(HERE, ['resources/']))
AssertionError: 'resources/empty.ipynb' not found in set()

======================================================================
FAIL: notebook.tests.test_notebookapp.test_nb_dir_with_slash
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\ADMINI~1\AppData\Local\Temp\2\28634\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\jenkins\workspace\notebook-windows\_pyversion\3.5\label\windows\notebook\tests\test_notebookapp.py", line 70, in test_nb_dir_with_slash
    nt.assert_false(app.notebook_dir.endswith("/"))
AssertionError: True is not false
```

--- a/notebook/bundler/tests/test_bundler_tools.py
+++ b/notebook/bundler/tests/test_bundler_tools.py
@@ -84,12 +84,12 @@ b/
 
     def test_glob_dir(self):
         '''Should expand to single file in the resources/ subfolder.'''
-        self.assertIn('resources/empty.ipynb',
+        self.assertIn(os.path.join('resources', 'empty.ipynb'),
             tools.expand_references(HERE, ['resources/empty.ipynb']))
 
     def test_glob_subdir(self):
         '''Should expand to all files in the resources/ subfolder.'''
-        self.assertIn('resources/empty.ipynb',
+        self.assertIn(os.path.join('resources', 'empty.ipynb'),
             tools.expand_references(HERE, ['resources/']))
 
     def test_glob_splat(self):
@@ -101,24 +101,24 @@ b/
     def test_glob_splatsplat_in_middle(self):
         '''Should expand to test_file.txt deep under this test/ directory.'''
         globs = tools.expand_references(HERE, ['resources/**/test_file.txt'])
-        self.assertIn('resources/subdir/test_file.txt', globs, globs)
+        self.assertIn(os.path.join('resources', 'subdir', 'test_file.txt'), globs, globs)
 
     def test_glob_splatsplat_trailing(self):
-        '''Should expand to all descendants of this test/ directory.''' 
+        '''Should expand to all descendants of this test/ directory.'''
         globs = tools.expand_references(HERE, ['resources/**'])
-        self.assertIn('resources/empty.ipynb', globs, globs)
-        self.assertIn('resources/subdir/test_file.txt', globs, globs)
+        self.assertIn(os.path.join('resources', 'empty.ipynb'), globs, globs)
+        self.assertIn(os.path.join('resources', 'subdir', 'test_file.txt'), globs, globs)
 
     def test_glob_splatsplat_leading(self):
         '''Should expand to test_file.txt under any path.'''
         globs = tools.expand_references(HERE, ['**/test_file.txt'])
-        self.assertIn('resources/subdir/test_file.txt', globs, globs)
-        self.assertIn('resources/another_subdir/test_file.txt', globs, globs)
+        self.assertIn(os.path.join('resources', 'subdir', 'test_file.txt'), globs, globs)
+        self.assertIn(os.path.join('resources', 'another_subdir', 'test_file.txt'), globs, globs)
 
     def test_copy_filelist(self):
         '''Should copy select files from source to destination'''
         globs = tools.expand_references(HERE, ['**/test_file.txt'])
         tools.copy_filelist(HERE, self.tmp, globs)
-        self.assertTrue(os.path.isfile(os.path.join(self.tmp, 'resources/subdir/test_file.txt')))
-        self.assertTrue(os.path.isfile(os.path.join(self.tmp, 'resources/another_subdir/test_file.txt')))
-        self.assertFalse(os.path.isfile(os.path.join(self.tmp, 'resources/empty.ipynb')))
+        self.assertTrue(os.path.isfile(os.path.join(self.tmp, 'resources', 'subdir', 'test_file.txt')))
+        self.assertTrue(os.path.isfile(os.path.join(self.tmp, 'resources', 'another_subdir', 'test_file.txt')))
+        self.assertFalse(os.path.isfile(os.path.join(self.tmp, 'resources', 'empty.ipynb')))

--- a/notebook/bundler/tools.py
+++ b/notebook/bundler/tools.py
@@ -11,8 +11,8 @@ import glob
 
 def get_file_references(abs_nb_path, version):
     """Gets a list of files referenced either in Markdown fenced code blocks
-    or in HTML comments from the notebook. Expands patterns expressed in 
-    gitignore syntax (https://git-scm.com/docs/gitignore). Returns the 
+    or in HTML comments from the notebook. Expands patterns expressed in
+    gitignore syntax (https://git-scm.com/docs/gitignore). Returns the
     fully expanded list of filenames relative to the notebook dirname.
 
     Parameters
@@ -21,7 +21,7 @@ def get_file_references(abs_nb_path, version):
         Absolute path of the notebook on disk
     version: int
         Version of the notebook document format to use
-    
+
     Returns
     -------
     list
@@ -41,7 +41,7 @@ def get_reference_patterns(abs_nb_path, version):
         Absolute path of the notebook on disk
     version: int
         Version of the notebook document format to use
-    
+
     Returns
     -------
     list
@@ -66,7 +66,7 @@ def get_cell_reference_patterns(cell):
     !foo/bar
     ```
 
-    or 
+    or
 
     <!--associate:
     some.csv
@@ -78,7 +78,7 @@ def get_cell_reference_patterns(cell):
     ----------
     cell: dict
         Notebook cell object
-    
+
     Returns
     -------
     list
@@ -113,29 +113,41 @@ def get_cell_reference_patterns(cell):
 
 def expand_references(root_path, references):
     """Expands a set of reference patterns by evaluating them against the
-    given root directory. Expansions are performed against patterns 
-    expressed in the same manner as in gitignore 
+    given root directory. Expansions are performed against patterns
+    expressed in the same manner as in gitignore
     (https://git-scm.com/docs/gitignore).
-    
+
     NOTE: Temporarily changes the current working directory when called.
 
     Parameters
     ----------
-    root_path: str 
+    root_path: str
         Assumed root directory for the patterns
     references: list
-        Reference patterns from get_reference_patterns
-        
+        Reference patterns from get_reference_patterns expressed with
+        forward-slash directory separators
+
     Returns
     -------
     list
         Filename strings relative to the root path
     """
+    # Use normpath to convert to platform specific slashes, but be sure
+    # to retain a trailing slash which normpath pulls off
+    normalized_references = []
+    for ref in references:
+        normalized_ref = os.path.normpath(ref)
+        # un-normalized separator
+        if ref.endswith('/'):
+            normalized_ref += os.sep
+        normalized_references.append(normalized_ref)
+    references = normalized_references
+
     globbed = []
     negations = []
     must_walk = []
     for pattern in references:
-        if pattern and pattern.find('/') < 0:
+        if pattern and pattern.find(os.sep) < 0:
             # simple shell glob
             cwd = os.getcwd()
             os.chdir(root_path)
@@ -156,7 +168,7 @@ def expand_references(root_path, references):
         for root, _, filenames in os.walk(root_path):
             for filename in filenames:
                 joined = os.path.join(root[len(root_path) + 1:], filename)
-                if testpattern.endswith('/'):
+                if testpattern.endswith(os.sep):
                     if joined.startswith(testpattern):
                         if pattern_is_negation:
                             negations.append(joined)
@@ -194,9 +206,9 @@ def copy_filelist(src, dst, src_relative_filenames):
 
     Parameters
     ----------
-    src: str 
+    src: str
         Root of the source directory
-    dst: str 
+    dst: str
         Root of the destination directory
     src_relative_filenames: list
         Filenames relative to src

